aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -76,9 +76,8 @@ module aptos_framework::coin {
         value: u64,
     }
 
-    /// Represents a coin with aggregator as its value. This allows to update
-    /// the coin in every transaction avoiding read-modify-write conflicts. Only
-    /// used for gas fees distribution by Aptos Framework (0x1).
+    #[deprecated]
+    /// DEPRECATED
     struct AggregatableCoin<phantom CoinType> has store {
```

Comment:
If this is truly private that only bypass function calls can access it, it might be, but that's such a special exception, I can see why no one would code for that case :P

Reply:
it is not safe to remove private struct that has "store", as you would be able to create a new one with different fields, and deserialize into it - breaking all type invariants

Reply:
> it is not safe to remove private struct that has "store", as you would be able to create a new one with different fields, and deserialize into it - breaking all type invariants

I think my original statement was in context to it being part of the framework, but the general statement is definitely true! Type confusion can happen sooo easily...

Reply:
Yes, that's why it seemed safe for me to remove: it is part of framework, and has private APIs, so even if you serialise into it what do you do after?